### ISSUE TITLE
Feature request: Convenience functions for primitive comparisons

### ISSUE THREAD

**Issue by cllns** [dillonkearns/elm-graphql#604]
I find myself using this very often: 
```elm
Api.InputObject.buildInt_comparison_exp (\args -> { args | eq_ = Present 123}
```

I understand this is powerful and enables complex queries, but it's also verbose for a trivial comparison.

I wonder if the basic primitives (Int, String, Date, etc), could have generated convenience functions. E.g. the above could leverage a convenience function like:

```elm
Api.Int.eq_ 123
```

The same would be repeated for the other operators: not equals, greater than (or equal to), less than (or equal to), is null, in list, not in list. If you need to combine them, then you need to fall-back to using `buildInt_comparison_exp`, but in the majority of cases, I'm doing simple comparisons that could be briefer.

These could be optionally generated (at least at first, to gain feedback) in the `elm-graphql` CLI command.

**Comment by cllns** [comment]
After sleeping on it, I realized I conflated Graphql with Hasura here 😅 

While Hasura is very popular within the Elm community, I guess you'd be reluctant (at best) to build anything that specifically generates code for that one provider within this general purpose library. Then, the obvious move would be to build a library that builds upon this one.

The simplest (zero integration) approach seems to be: create a tool to run _after_ `elm-graphql` that looks for `X.InputObject.buildInt_comparison_exp` (etc) with the right signature and then creates some files `XHelper.Int`. I think populating them in a different namespace is ideal, since I think this library wipes out all of the files each time it's run, is that right? (See below for a way around this constraint)

These primitive helper functions are relatively static so it doesn't necessarily need to be run _every_ time `elm-graphql` is run. However, adding new scalars to the schema, even primitive ones, means that new comparison expression functions get generated (e.g.`buildFloat_comparison_exp` when a float gets added to the schema). But since these are just convenience functions, that can be a manual additional step when needed.

Also new _custom_ scalars can be also get new comparison functions generated (e.g. `Uuid`, `Bpchar`), but those don't need to be considered for a first version, since those involve codecs.

Later, the `elm-hasura` CLI tool could first pass all its args to `elm-graphql` then runs its own pass immediately after that code is generated, rather than being a separate step. This would also let them share a namespace, `X.Int` as originally proposed, though I guess you'd prefer it if I stayed out of there thankyouverymuch 😉 

Unfortunately this all seems like a lot of work for a minor/moderate convenience. Are there any other obvious convenience function optimizations that could be made assuming the Graphql schema adheres to the Hasura conventions (or any given set of conventions)?


Otherwise: is there any appetite for adding extension points to this library to generate some extra files? Is that even possible? Maybe a `--after Hasura.GenerateHelpers.call` CLI hook? Ideally this would still have some state of the schema passed into it.

This same approach could be applied to creating convenience functions for [Postgraphile's similar tool](https://github.com/graphile-contrib/postgraphile-plugin-connection-filter), or really any Graphql schema.


I think in the meantime, I'll just make my own `ApiHelper` module with an `int_eq` function in my codebase as-needed :D
